### PR TITLE
Experimental script to enable workload identity.

### PR DESCRIPTION
**Pull request by duftler**

### CODE REVIEW

File: scripts/experimental/configure_for_workload_identity.sh [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+
+bold "Enabling GKE_METADATA_SERVER on node pool $NODE_POOL_NAME..."
+gcloud beta container node-pools update $NODE_POOL_NAME \
+  --cluster=$GKE_CLUSTER \
+  --zone=$ZONE \
+  --workload-metadata-from-node=GKE_METADATA_SERVER \
+  --project=$PROJECT_ID
+
+KSA_NAME=default
+GSA_NAME=spinnaker-wi-acct
```
Comment (maggieneterval) [inline scripts/experimental/configure_for_workload_identity.sh]: Just curious, what does "wi" stand for?

File: scripts/experimental/configure_for_workload_identity.sh [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+# The expected workflow is as follows:
+#   - Generate the properties file by running the setup_properties.sh script
+#   - Modify the properties file to specify the above 2 Spinnaker/Halyard versions (or later versions)
+#   - Run setup.sh
+#   - Once Spinnaker is up and running, run this (configure_for_workload_identity.sh) script
+#
+# Note that this script results in each Spinnaker pod still using the default Kubernetes service account,
+#   and the default service account in the halyard and spinnaker namespaces being bound to one Google
+#   service account (spinnaker-wi-acct). If you want to specify a different Kubernetes service account
+#   for any service, you can do so via the `servoiceAccountName` setting described here:
```
[the remark this answers is not in the document]
Comment (duftler) [reply scripts/experimental/configure_for_workload_identity.sh]: noice catch

File: scripts/experimental/configure_for_workload_identity.sh (hunk 1 above)
Comment (duftler) [reply scripts/experimental/configure_for_workload_identity.sh]: workload identity, or wisconsin 
image

File: scripts/experimental/configure_for_workload_identity.sh [hunk 3]
```diff
@@ -0,0 +1,152 @@
...
+#     will need to assign the appropriate roles to the spinnaker-wi-acct Google service account, similar
+#     to what we do in these helper scripts for the non-Workload Identity setup:
+#       https://github.com/GoogleCloudPlatform/spinnaker-for-gcp/blob/master/scripts/manage/add_gce_account.sh
+#       https://github.com/GoogleCloudPlatform/spinnaker-for-gcp/blob/master/scripts/manage/add_gae_account.sh
+
+bold() {
+  echo ". $(tput bold)" "$*" "$(tput sgr0)";
+}
+
+err() {
```
[the remark this answers is not in the document]
Comment (duftler) [reply scripts/experimental/configure_for_workload_identity.sh]: Was lazy with my copy/paste. Removed it.

File: scripts/experimental/configure_for_workload_identity.sh [hunk 4]
```diff
@@ -0,0 +1,148 @@
...
+bold "Enabling workload identity on cluster $GKE_CLUSTER in project $PROJECT_ID..."
+gcloud beta container clusters update $GKE_CLUSTER \
+  --zone=$ZONE \
+  --identity-namespace=$PROJECT_ID.svc.id.goog \
+  --project=$PROJECT_ID
+
+unset CLUSTER_STATUS
+
+while [ "$CLUSTER_STATUS" != "RUNNING" ]; do
+  CLUSTER_STATUS=$(gcloud beta container clusters describe $GKE_CLUSTER \
```
Comment (ezimanyi) [inline scripts/experimental/configure_for_workload_identity.sh]: Why are we using the `beta` command here?  (There are a few other places below as well that seem to be using GA functionality but are using the `beta` API surface.)

File: scripts/experimental/configure_for_workload_identity.sh [hunk 5]
```diff
@@ -0,0 +1,148 @@
...
+bold() {
+  echo ". $(tput bold)" "$*" "$(tput sgr0)";
+}
+
+PROPERTIES_FILE="$HOME/spinnaker-for-gcp/scripts/install/properties"
+
+source "$PROPERTIES_FILE"
+
+bold "Scaling down deployments to 0..."
+kubectl scale deployment spin-clouddriver -n spinnaker --replicas=0
```
Comment (ezimanyi) [inline scripts/experimental/configure_for_workload_identity.sh]: Why are we scaling these deployments down to 0 before we enable workload identity, then back up after?  The documentation on migrating workloads seems to imply that updating the node pool is enough.

Comment (duftler) [reply scripts/experimental/configure_for_workload_identity.sh]: The issue I ran into was that the pods are all rescheduled once the node pool is updated. This resulted in auth failures, and they never seemed to recover from this (it's possible this relates to not-yet-expired tokens. but I got tired of waiting).

I'm glad you asked about this as it forced me to work through it again. By moving the node pool update to the end, after the service accounts have been bound and the roles assigned, the auth failures are avoided and the scaling can be dropped. Thanks.

File: scripts/experimental/configure_for_workload_identity.sh (hunk 4 above)
Comment (duftler) [reply scripts/experimental/configure_for_workload_identity.sh]: Result of lazy copy/paste. It's not needed for the two `describe` queries, but is needed on the two mutating `update` calls (the workload identity specific flags are only supported via the `beta` API).

Done.

**Pull request merged**